sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java

```diff
@@ -446,6 +464,21 @@ public Node visitFunctionArgument(SqlBaseParser.FunctionArgumentContext context)
         return new FunctionArgument(getIdentTextIfPresent(context.ident()), (ColumnType) visit(context.dataType()));
     }
 
+    @Override
+    public Node visitPrivilege(SqlBaseParser.PrivilegeContext context) {
+        if (context.DML() != null){
+            return new Privilege(Privilege.PrivilegeType.DML);
+        }
+        if (context.DQL() != null){
+            return new Privilege(Privilege.PrivilegeType.DQL);
+        }
+
+        if (context.DDL() !=null){
+            return new Privilege(Privilege.PrivilegeType.DDL);
+        }
+        return null;
```

Comment:
should an error be thrown here when the privilege is not available ? or is it done at the analyser level ?